server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -242,6 +250,9 @@ protected static boolean parseObjectOrDocumentTypeProperties(
             } else if (fieldName.equals("enabled")) {
                 builder.enabled(XContentMapValues.nodeBooleanValue(fieldNode, fieldName + ".enabled"));
                 return true;
+            } else if (fieldName.equals("track_array_source")) {
```

Comment:
Maybe turn the name into a constant? I think `store_array_source`  maybe be better. This name is inline with the `store` param and the source mode when we store the source.

Maybe `store_source` is better if we like to extend this functionally for other fields as well?

Reply:
`store_array_source` soudns good. I think it's better to avoid a catch-all param since each one may have substantial impact on storage. We can add one in the future if we prefer so.